include/circt/Dialect/HW/HWAttributesNaming.td

```diff
@@ -16,16 +16,28 @@ def InnerRefAttr : AttrDef<HWDialect, "InnerRef"> {
     This works like a symbol reference, but to a name inside a module.
   }];
   let mnemonic = "innerNameRef";
-  let parameters = (ins "::mlir::StringAttr":$module, "::mlir::StringAttr":$name);
+  let parameters = (ins "::mlir::StringAttr":$module, "::mlir::StringAttr":$name,
+                        "unsigned":$fieldID);
   let builders = [
-    AttrBuilderWithInferredContext<(ins "::mlir::StringAttr":$module, "::mlir::StringAttr":$name),[{
-      return get(module.getContext(), module, name);
+    AttrBuilderWithInferredContext<(ins "::mlir::StringAttr":$module,
+                                        "::mlir::StringAttr":$name,
+                                        "unsigned":$fieldID),[{
```

Comment:
This should probably be an `I32Attr` or `APIntAttr` instead of an `unsigned`.

Reply:
It seems `I32Attr` or `APIntATttr` cannot be used as a parameter of attributes. I think it is possible to use mlir::IntegerAttr but  I feel there is no reason to use IntegerAttr here.